include/scsi-lowlevel.h

```diff
@@ -689,30 +689,84 @@ struct scsi_inquiry_device_identification {
 	struct scsi_inquiry_device_designator *designators;
 };
 
+/* Third-party copy descriptor type codes */
 enum third_party_copy_descriptor_type {
-	THIRD_PARTY_COPY_TYPE_SUPPORTED_COMMANDS  = 0x0001
+	THIRD_PARTY_COPY_TYPE_SUPPORTED_COMMANDS            = 0x0001,
+	THIRD_PARTY_COPY_TYPE_PARAMETER_DATA                = 0x0004,
+	THIRD_PARTY_COPY_TYPE_SUPPORTED_DESCRIPTORS         = 0x0008,
+	THIRD_PARTY_COPY_TYPE_SUPPORTED_CSCD_DESCRIPTORS_ID = 0x000c,
+	THIRD_PARTY_COPY_TYPE_GENERAL_COPY_OPERATIONS       = 0x8001
 };
 
+/* Command support descriptor */
 struct third_party_copy_command_support {
 	struct third_party_copy_command_support *next;
 
 	int operation_code;
-	int service_action_length;
-	int *service_action;
+	int service_actions_list_length;
+	int *service_actions;
 };
 
+/* Supported Commands third-party copy descriptor */
 struct third_party_copy_supported_commands {
 	enum third_party_copy_descriptor_type descriptor_type;
-
+	int descriptor_length;
+	int commands_supported_list_length;
 	struct third_party_copy_command_support *commands_supported;
 };
 
+/* Parameter Data third-party copy descriptor */
+struct third_party_copy_parameter_data {
+	enum third_party_copy_descriptor_type descriptor_type;
+	int descriptor_length;
+	unsigned char reserved_1[4];
+	int maximum_cscd_descriptor_count;
+	int maximum_segment_descriptor_count;
+	unsigned int maximum_descriptor_list_length;
+	unsigned int maximum_inline_data_length;
+	unsigned char reserved_2[12];
```

Comment:
Does this data structure come from a SCSI standard? If so, please use uint8_t, uint8_t, uint32_t etc. instead of int / unsigned char. If not, please remove the "reserved" members.